stdlib/public/Concurrency/DiscardingTaskGroup.swift

```diff
@@ -231,7 +244,20 @@ public struct DiscardingTaskGroup {
 #endif
 
     // Create the task in this group.
+#if $BuiltinCreateAsyncDiscardingTaskInGroup
     _ = Builtin.createAsyncDiscardingTaskInGroup(flags, _group, operation)
+#else
+    // This builtin happens to work, but the signature of the operation is
+    // incorrect, as the discarding group uses Void, and therefore has less
+    // generic parameters than the operation expected to be passed to
+    // createAsyncTaskInGroup. While this happened to work on some platforms,
+    // on others this causes issues, e.g. on wasm;
+    //
+    // Keep this branch for compatibility with old compilers, but use the
+    // correct 'createAsyncDiscardingTaskInGroup' when available (and a recent
+    // enough compiler is used).
+    _ = Builtin.createAsyncTaskInGroup(flags, _group, operation)
+#endif
```

Comment:
I think that's the right dance to do here -- the comment attempts to explain what we're doing here -- See the PR which introduced this change here: https://github.com/apple/swift/pull/70537/files#diff-9897ab704fe1b9e57499ec46e8837c683573998d299ea3cbfb64d29c4a34544aL567

So we're using what "used to work" on other platforms and old compilers, and when available, we use the "right thing".

Looks right @slavapestov @kavon ?